platforms/web/src/containers/AppRoutes/AppRoutes.tsx

```diff
@@ -54,7 +53,8 @@ export default function AppRoutes() {
   useNotifications();
 
   if (userData.user && !userData.loading && window.location.href.includes('#token')) {
-    return <Navigate to={`${PATH_HOME}${location.search}`} />; // component instead of hook to prevent extra re-renders
+    // leave the location intact and only remove the token from the hash
+    return <Navigate to={{ ...location, hash: '' }} replace />; // component instead of hook to prevent extra re-renders
```

Comment:
1. You can use the router's location instead of `window.location`
2. No need to pass in the previous location, only set want you want to change
3. Remove the entry from the history by adding `replace` 

```suggestion
  if (userData.user && !userData.loading && location.hash.includes('#token')) {
    // leave the location intact and only remove the token from the hash
    return <Navigate to={{ hash: '' }} replace />; // component instead of hook to prevent extra re-renders
  }
```